staging/src/k8s.io/client-go/util/certificate/certificate_manager.go

```diff
@@ -388,29 +388,11 @@ func (m *manager) rotateCerts() (bool, error) {
 	// Once we've successfully submitted a CSR for this template, record that we did so
 	m.setLastRequest(template)
 
-	// Wait for the certificate to be signed. Instead of one long watch, we retry with slightly longer
-	// intervals each time in order to tolerate failures from the server AND to preserve the liveliness
-	// of the cert manager loop. This creates slightly more traffic against the API server in return
-	// for bounding the amount of time we wait when a certificate expires.
-	var crtPEM []byte
-	watchDuration := time.Minute
-	if err := wait.ExponentialBackoff(certificateWaitBackoff, func() (bool, error) {
-		data, err := csr.WaitForCertificate(client, req, watchDuration)
-		switch {
-		case err == nil:
-			crtPEM = data
-			return true, nil
-		case err == wait.ErrWaitTimeout:
-			watchDuration += time.Minute
-			if watchDuration > 5*time.Minute {
-				watchDuration = 5 * time.Minute
-			}
-			return false, nil
-		default:
-			utilruntime.HandleError(fmt.Errorf("Unable to check certificate signing status: %v", err))
-			return false, m.updateServerError(err)
-		}
-	}); err != nil {
+	// Wait for the certificate to be signed. This interface and internal timout
+	// is a remainder after the old design using raw watch wrapped with backoff.
+	timeout := 15 * time.Minute
```

Comment:
hope that seems like a reasonable timeout - there was 1m + the backoff
this whole func is wrapped in backoff and poolInfinite in other place and it account for failures. It would be nice to plumb it through but that's not in scope of this PR.

Reply:
previously, the backoff was a private package var we manipulated in tests... can we do the same with the timeout, to restore TestRotateCertWaitingForResultError?

Reply:
(previously, the backoff took 4 * 1m + ~8 minutes + watch durations, so 15 minutes approximates current delay and seems ok)